packages/ncids-js/src/components/nci-header/utils/megamenu.component.ts

```diff
@@ -0,0 +1,557 @@
+import { NCIMegaMenuOptions } from './megamenu-options';
+import { NavItem } from './navigation-item';
+import { FocusTrap } from './focustrap';
+
+/**
+ * NCI MegaMenu
+ *
+ *
+ * Initialize the MegaMenu component:
+ * ```
+ * NCIMegaMenu.element.create(HTMLElement);
+ * ```
+ */
+export class NCIMegaMenu {
+	/** The component that contains megamenu. */
+	protected element: HTMLElement;
+	/** The options for MegaMenu */
+	protected options: NCIMegaMenuOptions;
+	/** The Primary navigation buttons for the Megamenu */
+	private toggleButtons: HTMLElement[] = [];
+	/** Focus Trap class for Mega Menus*/
+	private focusTrap: FocusTrap;
+	/** Array list of custom events that will be dispatched to the user. */
+	private customEvents: { [key: string]: CustomEvent } = {};
+	/** Callback for handling the toggle functionality.  */
+	private eventListener: EventListener = (event) => this.handleClick(event);
+	/** Callback for handling clicking off the menus.  */
+	private eventOffListener: EventListener = (event) =>
+		this.handleClickOff(event);
+	/** Callback for keyboard input for a11y.  */
+	private keyboardListener: EventListener = (event: Event) =>
+		this.handleKeyboard(<KeyboardEvent>event);
+	/** Is the MegaMenu active  */
+	private isActive: boolean;
+	/** Default classNames for the component. */
+	private static defaultOptions: NCIMegaMenuOptions = {
+		listElementClass: '.nci-header-nav__primary-item',
+		linkElementClass: '.nci-header-nav__primary-link',
+	};
+
+	/** Map object of the component. */
+	private static _components: Map<HTMLElement, NCIMegaMenu> = new Map<
+		HTMLElement,
+		NCIMegaMenu
+	>();
+
+	/**
+	 * Sets component variables and initializes component.
+	 *
+	 * @param {HTMLElement} element Component being created.
+	 * @param {Partial<NCIMegaMenuOptions>} options Component being created.
+	 * @protected
+	 */
+	protected constructor(element: HTMLElement, options: NCIMegaMenuOptions) {
+		this.element = element;
+		this.options = {
+			...NCIMegaMenu.defaultOptions,
+			...options,
+		};
+		this.focusTrap = new FocusTrap(this.element);
+		this.isActive = false;
+		const existingComponent = NCIMegaMenu._components.get(this.element);
+		if (existingComponent) {
+			existingComponent.unregister();
+		}
+
+		this.initialize();
+		NCIMegaMenu._components.set(this.element, this);
+	}
+
+	/**
+	 * Instantiates this component of the given element.
+	 *
+	 * @param {HTMLElement} element Component being created.
+	 * @param {Partial<NCIMegaMenuOptions>} options Component being created.
+	 * @public
+	 */
+	public static create(
```

Comment:
i don't think we want any create methods on util classes, an unnecessary layer as we should not call megamenu outside of the header. Of course, you'll have to update the NCIHeader class and all the tests..